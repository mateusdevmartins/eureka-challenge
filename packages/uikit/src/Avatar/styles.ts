import styled from 'styled-components';

export const AvatarImage = styled.img`
  align-self: center;
  display: flex;
  align-items: center
  margin: 0px 8px 0px 8px;
`;

export const AvatarView = styled.div`
  align-self: center;
  display: flex;
  align-items: center
  margin: 0px 8px 0px 8px;
  border-width: 1px;
  border-style: solid;
  border-color: #16325c;
`;
