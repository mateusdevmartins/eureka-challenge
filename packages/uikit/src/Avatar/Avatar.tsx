import { FaUserAlt } from 'react-icons/fa';
import { AvatarImage, AvatarView } from './styles';

export interface AvatarProps {
  image?: string | undefined;
  size: string;
}

export function Avatar({ image, size }: AvatarProps) {
  if (image) {
    return (
      <AvatarImage
        data-testid="avatar"
        src={image}
        alt="Avatar"
        style={{
          width: size,
          height: size,
          borderRadius: size,
        }}
      />
    );
  } else {
    return (
      <AvatarView
        data-testid="empty-avatar"
        style={{
          width: size,
          height: size,
          borderRadius: size,
        }}
      >
        <FaUserAlt size={size} style={{ margin: '0px 8px 0px 8px' }} />
      </AvatarView>
    );
  }
}
