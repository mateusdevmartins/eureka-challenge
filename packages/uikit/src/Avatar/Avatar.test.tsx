import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Avatar } from './Avatar';

describe('Avatar ', () => {
  const props = {
    image: 'image',
    size: '24px',
  };

  const props2 = {
    size: '24px',
  };
  it('should match snapshot', () => {
    const component = renderer.create(<Avatar {...props}></Avatar>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has image', () => {
    const { getByTestId } = render(<Avatar {...props} />);
    expect(getByTestId('avatar')).toBeTruthy();
    expect(() => getByTestId('empty-avatar')).toThrow(
      'Unable to find an element'
    );
  });

  it('has no image', () => {
    const { getByTestId } = render(<Avatar {...props2} />);
    expect(getByTestId('empty-avatar')).toBeTruthy();
    expect(() => getByTestId('avatar')).toThrow('Unable to find an element');
  });
});
