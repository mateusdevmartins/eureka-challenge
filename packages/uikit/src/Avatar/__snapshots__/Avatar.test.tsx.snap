// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar  should match snapshot 1`] = `
<img
  alt="Avatar"
  className="styles__AvatarImage-khcs16-0 ijOVLW"
  data-testid="avatar"
  src="image"
  style={
    Object {
      "borderRadius": "24px",
      "height": "24px",
      "width": "24px",
    }
  }
/>
`;
